docs/plugins/pytorch_dali_proxy.rst

```diff
@@ -0,0 +1,227 @@
+
+.. code-block:: python
+
+   from torchvision import datasets, transforms
+   from nvidia.dali import pipeline_def, fn, types
+   from nvidia.dali.plugin.pytorch.experimental import proxy as dali_proxy
+
+   # Step 1: Define a DALI pipeline
+   @pipeline_def
+   def my_dali_pipeline():
+       images = fn.external_source(name="images", no_copy=True)
+       images = fn.resize(images, size=[224, 224])
+       return fn.crop_mirror_normalize(
+           images, dtype=types.FLOAT, output_layout="CHW",
+           mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
+           std=[0.229 * 255, 0.224 * 255, 0.225 * 255],
+       )
+
+   # Step 2: Initialize DALI server. The scope makes sure to start and stop the background thread
+   with dali_proxy.DALIServer(my_dali_pipeline(batch_size=64, num_threads=3, device_id=0)) as dali_server:
+       # Step 3: Define a PyTorch Dataset using the DALI proxy
+       dataset = datasets.ImageFolder("/path/to/images", transform=dali_server.proxy)
+       
+       # Step 4: Use DALI proxy DataLoader
+       loader = dali_proxy.DataLoader(dali_server, dataset, batch_size=64, num_workers=8, drop_last=True)
+       
+       # Step 5: Consume data
+       for data, target in loader:
+           print(data.shape)  # Processed data ready
+
+How It Works
+------------
+
+**1. DALI Pipeline**
+
+The DALI pipeline defines the data processing steps. Input data is fed using :meth:`~nvidia.dali.fn.external_source`.
+
+.. code-block:: python
+
+   from nvidia.dali import pipeline_def, fn, types
+
+   @pipeline_def
+   def example_pipeline():
+       images = fn.external_source(name="images", no_copy=True)
+       images = fn.io.file.read(images)
+       images = fn.decoders.image(images, device="mixed", output_type=types.RGB)
+       return fn.resize(images, size=[224, 224])
+
+   pipeline = example_pipeline(batch_size=32, num_threads=2, device_id=0)
+
+**2. DALI Server and Proxy**
+
+The :class:`nvidia.dali.plugin.pytorch.experimental.proxy.DALIServer` manages the execution of the pipeline. The Proxy acts 
+as an interface for PyTorch data workers.
+Note that the DALI pipeline should contain at least one input (an :meth:`~nvidia.dali.fn.external_source` instance), and
+that the names of those nodes then become the inputs to the DALI proxy callable.
+
+.. code-block:: python
+
+   from nvidia.dali.plugin.pytorch.experimental import proxy as dali_proxy
+   with dali_proxy.DALIServer(pipeline) as dali_server:
+      future_samples = [dali_server.proxy(image) for image in images]
+
+With more than one input, we can choose to use positional arguments, keyword arguments:
+
+.. code-block:: python
+
+   import numpy as np
+   from nvidia.dali import pipeline_def, fn, types
+   from nvidia.dali.plugin.pytorch.experimental import proxy as dali_proxy
+
+   @pipeline_def
+   def example_pipeline2(device):
+      a = fn.external_source(name="a", no_copy=True)
+      b = fn.external_source(name="b", no_copy=True)
+      return a + b, b - a
```

Comment:
```suggestion
      return b - a, a + b
```
Kaboom, now the pipeline input names are `b', `'a'` and the order of positional arguments is swapped.